packages/manager/modules/pci/src/projects/project/public-gateways/add/add.controller.js

```diff
@@ -54,6 +72,52 @@ export default class PciPublicGatewaysAddController {
     };
   }
 
+  getDefaultValue(field) {
+    return (
+      this.defaults[field.name] !== undefined &&
+      isFunction(field.getDefault) &&
+      field.getDefault(this.defaults[field.name], this[field.availableOptions])
+    );
+  }
+
+  onChangeAfterDefaultValueSet(methodToCall, params) {
+    if (this[methodToCall] && isFunction(this[methodToCall])) {
+      return this[methodToCall].call(
+        this,
+        ...params.map((prop) => get(this, prop)),
+      );
+    }
+    return null;
+  }
+
+  async setDefaultSelections() {
+    this.loadingDefaultValues = true;
+    for (let i = 0; i < DEFAULTS_MODEL.length; i += 1) {
+      const step = DEFAULTS_MODEL[i];
+      for (let j = 0; j < step.fields.length; j += 1) {
+        const defaultvalue = this.getDefaultValue(step.fields[j]);
+        if (defaultvalue !== undefined) {
+          set(this, step.fields[j].model, defaultvalue);
+          // eslint-disable-next-line no-await-in-loop
+          await this.onChangeAfterDefaultValueSet(
+            step.fields[j].onChange,
+            step.fields[j].onChangeParams,
+          );
+        } else {
+          this.$timeout(() => {
+            this.currentStep = i;
+            this.loadingDefaultValues = false;
+          });
+          return;
+        }
+      }
+      this.currentStep = i;
+    }
+    this.$timeout(() => {
+      this.loadingDefaultValues = false;
+    });
+  }
```

Comment:
**suggestion:** Since the same logic is also used in "Addition IP" section. Can we move it to utility instead of duplicating the code?

Reply:
@anooparveti moved this as utility.